crates/rspack_binding_api/src/plugins/interceptor.rs

```diff
@@ -1456,7 +1453,7 @@ impl NormalModuleFactoryBeforeResolve for NormalModuleFactoryBeforeResolveTap {
       .await
     {
       Ok((ret, resolve_data)) => {
-        dependency.set_request(resolve_data.request);
```

Comment:
Seems this is the only usage about set_request, can we remove set_request on Dependency trait now?